### PR TITLE
[TW-6557] Pack size

### PR DESCRIPTION
**Pull request by NathanDeVuono**
Adds the pack selector, as well as making pack availability and subscription availability data driven.

### Screenshots
Mobile:
image

Desktop:
image

### CODE REVIEW

File: components/product/ProductView/ProductView.tsx [hunk 1]
```diff
@@ -28,9 +33,22 @@ interface Props {
...
+          +o.id === SUBSCRIPTIONTYPE_ID &&
+          o.values.filter((v) => v.label !== 'None').length > 0
+      ).length > 0
+  )
+  const hasPremium = premiumVariants.length > 0
+  const hasProtect =
+    premiumVariants.filter(
+      (v) =>
+        v.options.filter((o) => o.values.filter((v) => v.label !== 'Premium'))
+          .length > 0
```
Comment (tomasztile) [inline components/product/ProductView/ProductView.tsx:50]: is the ` > 0` needed for a truthy check?

File: components/product/ProductView/ProductView.tsx [hunk 2]
```diff
@@ -28,9 +33,22 @@ interface Props {
...
-  const [productDetails] = useState(JSON.parse(product.warranty))
-  const [hasPremium] = [true]
-  const [hasProtect] = [false]
+  const productDetails = JSON.parse(product.warranty)
+  const premiumVariants = product.variants.filter(
+    (v) =>
+      v.options.filter(
+        (o) =>
+          +o.id === SUBSCRIPTIONTYPE_ID &&
+          o.values.filter((v) => v.label !== 'None').length > 0
```
Comment (tomasztile) [inline components/product/ProductView/ProductView.tsx:42]: Is the `> 0` needed for a truthy check?

File: components/product/ProductView/ProductView.tsx [hunk 3]
```diff
@@ -102,11 +119,42 @@ const ProductView: FC<Props> = ({ product, variants }) => {
...
+    setChoices({
+      ...choices,
+      'pack size': e.target.attributes.getNamedItem('value')!.value,
+    })
+  }
+
+  const availability = product.variants.reduce((acc: AvailabilityObject, v) => {
+    const packSize = +v.options.filter((o) => +o.id === PACK_SIZE_ID)[0]
+      ?.values[0].label
+    if (packSize === 1) {
```
Comment (tomasztile) [inline components/product/ProductView/ProductView.tsx:135]: Perhaps use a switch statement.

File: components/product/ProductView/ProductView.tsx [hunk 4]
```diff
@@ -102,11 +119,42 @@ const ProductView: FC<Props> = ({ product, variants }) => {
...
+    }
+    if (packSize === 8) {
+      acc.eightPack = v.inventory.isInStock
+    }
+    return acc
+  }, {})
+
+  const pageVariant = variants.find((v) => v.id === product.id)!
+
+  console.log('Choices: ', choices)
```
Comment (tomasztile) [inline components/product/ProductView/ProductView.tsx:152]: Are we keeping these log statements?

**Review by tomasztile — APPROVED** [review]

File: components/product/ProductView/ProductView.tsx (hunk 4 above)
Comment (NathanDeVuono) [reply components/product/ProductView/ProductView.tsx:152]: They'll get ripped out when the PDP is actually done

File: components/product/ProductView/ProductView.tsx (hunk 1 above)
Comment (NathanDeVuono) [reply components/product/ProductView/ProductView.tsx:50]: Not strictly, I could double-bang the result, but I felt this was a little more explicit

Comment (tomasztile) [reply components/product/ProductView/ProductView.tsx:50]: I keep forgetting about TypeScript's strictness :P, this is better `> 0`

File: components/product/ProductView/ProductView.tsx (hunk 3 above)
Comment (NathanDeVuono) [reply components/product/ProductView/ProductView.tsx:135]: I always forget at what point a switch gets compiled down to `if`s, but you're right - a switch is easier to read. Will update

**Review by tomasztile — APPROVED** [review]

**Pull request merged**